packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js

```diff
@@ -25,16 +39,32 @@ export default function SidebarNavigationScreenGlobalStyles() {
 			) }
 			content={ <StyleVariationsContainer /> }
 			actions={
-				<SidebarButton
-					icon={ edit }
-					label={ __( 'Edit styles' ) }
-					onClick={ () => {
-						// switch to edit mode.
-						setCanvasMode( 'edit' );
-						// open global styles sidebar.
-						openGeneralSidebar( 'edit-site/global-styles' );
-					} }
-				/>
+				<div>
+					<SidebarButton
+						icon={ seen }
+						label={ __( 'Style Book' ) }
+						onClick={ () => {
+							// Switch to edit mode.
+							setCanvasMode( 'edit' );
+							// Open global styles sidebar.
+							openGeneralSidebar( 'edit-site/global-styles' );
+							// Open style book, via the useEffect above.
+							// This is done via a local state change to ensure that the canvas mode is changed,
+							// and the global styles sidebar is opened before attempting to open the style book.
+							setIsStyleBookOpened( true );
```

Comment:
I see why you did it this way, and it works well. 👍 

I think dispatches return a promise so you might not even need `useState`

```jsx
						onClick={ async () => {
							await setCanvasMode( 'edit' );
							await openGeneralSidebar(
								'edit-site/global-styles'
							);
							setEditorCanvasContainerView( 'style-book' )
						} }
```